System.Diagnostics.Tests.ProcessWaitingTests.WaitForPeerProcess failed in CI

Configuration: OuterLoop_Fedora24_debug ([build#119](https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_fedora24_debug/119/testReport/))

Detail: https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_fedora24_debug/119/testReport/System.Diagnostics.Tests/ProcessWaitingTests/WaitForPeerProcess/

~~~
MESSAGE:
Assert.Equal() Failure\nExpected: Signal\nActual: (null)
+++++++++++++++++++
STACK TRACE:
at System.Diagnostics.Tests.ProcessWaitingTests.WaitForPeerProcess() in /mnt/resource/j/workspace/dotnet_corefx/master/outerloop_fedora24_debug/src/System.Diagnostics.Process/tests/ProcessWaitingTests.cs:line 157
~~~

```
     System.Diagnostics.Tests.ProcessWaitingTests.WaitForPeerProcess [FAIL]
        Assert.Equal() Failure
        Expected: Signal
        Actual:   (null)
...
/mnt/resource/j/workspace/dotnet_corefx/master/fedora24_debug/Tools/tests.targets(326,5): warning MSB3073: The command "/mnt/resource/j/workspace/dotnet_corefx/master/fedora24_debug/bin/Unix.AnyCPU.Debug/System.Diagnostics.Process.Tests/netstandard/RunTests.sh /mnt/resource/j/workspace/dotnet_corefx/master/fedora24_debug/bin/testhost/netcoreapp-Linux-Debug-x64/" exited with code 1. [/mnt/resource/j/workspace/dotnet_corefx/master/fedora24_debug/src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests.csproj]
```

dumpling seems to has failed to download
```
  Traceback (most recent call last):
    File "dumpling.py", line 18, in <module>
      import requests
  ImportError: No module named requests
  python: can't open file '/mnt/resource/j/workspace/dotnet_corefx/master/fedora24_debug/tempHome/.dumpling/dumpling.py': [Errno 2] No such file or directory
  executing ulimit -c unlimited
  command exited with ExitCode: 1
  command failed, trying to collect dumps
  corefile:
  no coredump file was found
  /mnt/resource/j/workspace/dotnet_corefx/master/fedora24_debug/src/System.Diagnostics.Process/tests
  Finished running tests. End time=23:03:27. Return value was 1
```

--- a/src/System.Diagnostics.Process/tests/ProcessWaitingTests.cs
+++ b/src/System.Diagnostics.Process/tests/ProcessWaitingTests.cs
@@ -155,7 +155,9 @@ namespace System.Diagnostics.Tests
             }, child1.Id.ToString());
             child2.StartInfo.RedirectStandardOutput = true;
             child2.Start();
-            Assert.Equal("Signal", child2.StandardOutput.ReadLine()); // wait for the signal before killing the peer
+            char[] output = new char[6];
+            child2.StandardOutput.Read(output, 0, output.Length);
+            Assert.Equal("Signal", new string(output)); // wait for the signal before killing the peer
 
             child1.Kill();
             Assert.True(child1.WaitForExit(WaitInMS));

